### PR TITLE
Small fix and ignore some metafiles

### PR DESCRIPTION
**Pull request by jonaprieto**
Besides fixing the website with a small issue introduced by:

- #501,

after this PR, files generated automatically will be ignored, namely,

- MODULE-INDEX.md
- SUMMARY.md
- CONTRIBUTORS.md (which requires calls to the GitHub API).

Version: https://jonaprieto.github.io/agda-unimath

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -47,7 +47,7 @@ repos:
 
     - id: generate-main-index-file
       name: Generate main index file
-      entry: hooks/generate_main_index_file.py
+      entry: scripts/generate_main_index_file.py
       language: python
       always_run: true
       verbose: true
```
[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply .pre-commit-config.yaml:53]: Maybe move this one out of `pre-commit`

[the remark this answers is not in the document]
Comment (jonaprieto) [reply .pre-commit-config.yaml:59]: True

**Comment by fredrik-bakke** [comment]
Maybe the include issue can be fixed by simply including MODULE-INDEX into HOME before CONTRIBUTORS instead of including it into SUMMARY?

File: scripts/generate_main_index_file.py [hunk 2]
```diff
@@ -63,17 +65,42 @@ def generate_index(root, header):
...
+  - [How-to](HOWTO-INSTALL.md)
+    - [Install](HOWTO-INSTALL.md)
+    - [Cite the library](CITATION.cff)
+  - [Guidelines](CODINGSTYLE.md)
+    - [Structure your file](CONVENTIONS.md)
+    - [Library coding style](CODINGSTYLE.md)
+    - [Design principles](DESIGN-PRINCIPLES.md)
+  - [Everything](everything.md)
+
+MODULE_INDEX
```
Comment (fredrik-bakke) [inline scripts/generate_main_index_file.py:90]: This token should be replaced with usage of a format string token `{MODULE_INDEX}`

File: scripts/generate_namespace_index_modules.py [hunk 3]
```diff
@@ -1,23 +1,28 @@
...
+
 def generate_title(namespace):
     return "# " + namespace.capitalize().replace("-", " ") + "\n"
 
+
 def generate_imports(root, namespace):
     namespace_path = os.path.join(root, namespace)
-    agda_file_filter = lambda f: utils.isAgdaFile(pathlib.Path(os.path.join(namespace_path, f)))
+    def agda_file_filter(f): return utils.isAgdaFile(
+        pathlib.Path(os.path.join(namespace_path, f)))
```
Comment (fredrik-bakke) [inline scripts/generate_namespace_index_modules.py:18]: You don't like lambdas?

**Review by fredrik-bakke — COMMENTED** [review]
Only had a minor comment. Great PR! 😄

File: scripts/generate_main_index_file.py [hunk 4]
```diff
@@ -63,17 +65,42 @@ def generate_index(root, header):
...
     index_header = "# Formalisation in Agda"
 
     index_content, status = generate_index(root, index_header)
-
-    with open(index_path, "w") as index_file:
-        index_file.write(index_content)
-
+    if status == 0:
+        summary_contents = summary_template.replace(
+            "MODULE_INDEX", index_content)
```
[the remark this answers is not in the document]
Comment (jonaprieto) [reply scripts/generate_main_index_file.py:103]: We should do this way in the next PR.

File: scripts/generate_namespace_index_modules.py (hunk 3 above)
Comment (jonaprieto) [reply scripts/generate_namespace_index_modules.py:18]: I don't remember this. Please change as you think in your next PR.
I do like lambdas.

**Pull request merged**